openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java

```diff
@@ -94,6 +94,9 @@ public void before(BasicBuilder builder, HttpRequest httpRequest, RequestTags ta
 
   @Override
   public CompletableFuture<Boolean> afterFailure(Builder builder, HttpResponse<?> response, RequestTags tags) {
+    if (config.isTokenNonRefreshable()) {
+      return CompletableFuture.completedFuture(true);
```

Comment:
Returning true here and in the other interceptor means to try the request again, but since no modifications have been made on the builder, it would just re-issue the existing request.

Including the token provider in isTokenNonRefreshable could be confusing as it may refresh.  That is:

if token source == user, then do nothing
if token source == provider, then attempt to get the token again and try once more.